doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -237,202 +236,153 @@ analysis on inductive or co-inductive objects (see :ref:`variants`).
 Induction
 ---------
 
-.. tacn:: induction @term
-   :name: induction
+.. tacn:: induction {+, @induction_clause } {? @induction_principle }
+
+   .. insertprodn induction_principle induction_principle
+
+   .. prodn::
+      induction_principle ::= using @one_term {? with @bindings } {? @occurrences }
 
-   This tactic applies to any goal. The argument :n:`@term` must be of
-   inductive type and the tactic :n:`induction` generates subgoals, one for
-   each possible form of :n:`@term`, i.e. one for each constructor of the
-   inductive type.
+   Applies an :term:`induction principle` to generate a subgoal for
+   each constructor of an inductive type.
 
-   If the argument is dependent in either the conclusion or some
+   If the argument is :term:`dependent <dependent product>` in the conclusion or some
    hypotheses of the goal, the argument is replaced by the appropriate
-   constructor form in each of the resulting subgoals and induction
+   constructor in each of the resulting subgoals and induction
    hypotheses are added to the local context using names whose prefix
-   is **IH**.
-
-   There are particular cases:
+   is **IH**.  The tactic is similar to :tacn:`destruct`, except that
+   `destruct` doesn't generate induction hypotheses.
 
-   + If term is an identifier :n:`@ident` denoting a quantified variable of the
-     conclusion of the goal, then inductionident behaves as :n:`intros until
-     @ident; induction @ident`. If :n:`@ident` is not anymore dependent in the
-     goal after application of :n:`induction`, it is erased (to avoid erasure,
-     use parentheses, as in :n:`induction (@ident)`).
-   + If :n:`@term` is a :n:`@natural`, then :n:`induction @natural` behaves as
+   :n:`induction @one_term ...` also works when the type of :n:`@one_term` is a statement
+   with premises and whose conclusion is inductive. In that case the tactic
+   performs induction on the conclusion of the type of :n:`@one_term` and leaves the
+   :term:`non-dependent premises <dependent premise>` of the type as subgoals.
+   In the case of dependent products, the tactic tries to find an instance for
+   which the induction principle applies and otherwise fails.
+
+   Most parameters behave as they do in :tacn:`destruct`, with the following
+   differences:
+
+   + If :n:`@one_term` (in :n:`@induction_arg`)
+     is an identifier :n:`@ident` denoting a quantified variable of the
+     conclusion of the goal, then :n:`induction @ident` behaves like :tacn:`intros` :n:`until
```

Comment:
e.g., refers to `B` when the goal is `A -> B`?

Reply:
In this case, I'm afraid that's the other meaning, i.e. the part below the `--------`.